src/main/java/uk/gov/pay/api/model/CreateCardPaymentRequest.java

```diff
@@ -197,19 +197,34 @@ public Optional<Internal> getInternal() {
     }
 
     @JsonProperty("set_up_agreement")
-    @Schema(description = "ID of agreement to save card details to", required = false, example = "abcefghjklmnopqr1234567890", hidden = true)
+    @Schema(description = "Use this parameter to set up an existing agreement for recurring payments. " +
+            "The `set_up_agreement` value you send must be a valid `agreement_id`.", 
+            required = false, example = "abcefghjklmnopqr1234567890")
     public Optional<String> getSetUpAgreement() {
         return Optional.ofNullable(setUpAgreement);
     }
 
     @JsonProperty("agreement_id")
-    @Schema(description = "ID of agreement to take payment from (requires \"authorisation_mode\": \"agreement\")", required = false, example = "abcefghjklmnopqr1234567890", hidden = true)
+    @Schema(description = "The unique ID GOV.UK Pay automatically associated with a recurring payments agreement. " +
+            "Including `agreement_id` in your request tells the API to take this payment using the card details that are associated with this agreement. " +
+            "`agreement_id` must match an active agreement ID. " +
+            "You must set `authorisation_mode` to `agreement` for the API to accept `agreement_id`.", 
+            required = false, example = "abcefghjklmnopqr1234567890")
     public Optional<String> getAgreementId() {
         return Optional.ofNullable(agreementId);
     }
 
     @JsonProperty("authorisation_mode")
-    @Schema(hidden = true)
+    @Schema(description = "Sets how you intend to authorise the payment. Defaults to `web`. " +
+            "Payments created with `web` mode follow the [standard GOV.UK Pay payment journey](https://docs.payments.service.gov.uk/payment_flow/). " +
+            "Paying users visit the `next_url` in the response to complete their payment. " +
+            "Payments created with `agreement` mode are authorised with an agreement for recurring payments. " +
+            "If you create an `agreement` payment, you must also send an active `agreement_id`. " +
+            "You must not send `return_url`, `email`, or `prefilled_cardholder_details` or your request will fail. " +
+            "Payments created with `moto_api` mode return an `auth_url_post` object and a `one_time_token`. " +
+            "You can use `auth_url_post` and `one_time_token` to send the paying user’s card details through the API and complete the payment. " +
+            "If you create a `moto_api` payment, do not send a `return_url` in your request.",
+            allowableValues = {"web", "agreement", "moto_api"})
```

Comment:
I think the allowableValues and the enum values are both being picked up when generating the `enum` in the OpenApi spec.

We only want to include this subset of values as `external` is not a valid value for a create payment request, but it can be returned in responses to get a payment. This is because payments with an authorisation_mode of `external` are created through a different API.

Not quite sure how we only show the subset for this API endpoint though..

Reply:
Looks like this is the solution https://github.com/springdoc/springdoc-openapi/issues/1243

`@Schema(type = "String", ...`